storage/src/vespa/storage/visiting/visitor.cpp

```diff
@@ -511,36 +495,46 @@ Visitor::start(api::VisitorId id, api::StorageMessage::Id cmdId,
         _buckets.size(),
         _visitorOptions._fromTime.getTime(),
         _visitorOptions._toTime.getTime(),
-        (buckets.size() > 0 ? _buckets[0].toString().c_str() : ""),
+        (!buckets.empty() ? _buckets[0].toString().c_str() : ""),
         _visitorOptions._maxPending,
         (_visitorOptions._visitRemoves ? "true" : "false"),
         _visitorOptions._fieldSet.c_str());
 }
 
+namespace {
+
+vespalib::steady_time
+capped_future(vespalib::steady_time time, vespalib::duration duration) {
+    vespalib::steady_time future = time + duration;
+    return (future < time)
+        ? vespalib::steady_time::max()
+        : future;
+}
+
+}
+
 void
 Visitor::attach(std::shared_ptr<api::CreateVisitorCommand> initiatingCmd,
                 const mbus::Route& controlAddress,
                 const mbus::Route& dataAddress,
-                framework::MilliSecTime timeout)
+                vespalib::duration timeout)
 {
     _priority = initiatingCmd->getPriority();
-    _timeToDie = _component.getClock().getTimeInMicros() + timeout.getMicros();
+    _timeToDie = capped_future(_component.getClock().getMonotonicTime(), timeout);
```

Comment:
The issue is that we some places have set the timeout as default to vespalib::duration::max(),
See https://github.com/vespa-engine/vespa/blob/18a5576a8e1b295f9bdbec55270ebdcb7e0ca05e/storage/src/vespa/storageapi/messageapi/storagecommand.cpp#L21
And that overflows giving time in the past....

Reply:
`std::chrono` time point and duration types use a signed representation internally, so an eager compiler will happily remove the branch in `capped_future` since signed overflows are undefined behavior and thus Cannot Happen™. So another mechanism will have to be used to detect overflows. Perhaps explicitly special-casing the `duration::max` timeout.

Reply:
@havardpe FYI